hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/utils/QuotaHelper.java

```diff
@@ -0,0 +1,137 @@
+     * @param parentType
+     *            The type of the parent entity.
+     * 
+     * @throws QuotaExceededException
+     *             if the assignment operation would cause the quota to be
+     *             exceeded
+     */
+    public static void assertAssignmentQuota(final long requested, final long limit, @NotNull final Class<?> type,
+            @NotNull final Class<?> parentType) {
+        // dispatch
+        assertAssignmentQuota(null, requested, limit, type.getSimpleName(), parentType.getSimpleName(), null);
+    }
+
+    /**
+     * Asserts the specified assignment quota.
+     * 
+     * @param parentId
+     *            The ID of the parent entity.
+     * @param requested
+     *            The number of entities that shall be assigned to the parent
+     *            entity.
+     * @param limit
+     *            The maximum number of entities that may be assigned to the
+     *            parent entity.
+     * @param type
+     *            The type of the entities that shall be assigned.
+     * @param parentType
+     *            The type of the parent entity.
+     * @param countFct
+     *            Function to count the entities that are currently assigned to
+     *            the parent entity.
+     * 
+     * @throws QuotaExceededException
+     *             if the assignment operation would cause the quota to be
+     *             exceeded
+     */
+    public static void assertAssignmentQuota(final Long parentId, final long requested, final long limit,
+            @NotNull final Class<?> type, @NotNull final Class<?> parentType, final Function<Long, Long> countFct) {
+        // dispatch
+        assertAssignmentQuota(parentId, requested, limit, type.getSimpleName(), parentType.getSimpleName(), countFct);
+    }
+
+    /**
+     * Asserts the specified assignment quota.
+     * 
+     * @param parentId
+     *            The ID of the parent entity.
+     * @param requested
+     *            The number of entities that shall be assigned to the parent
+     *            entity.
+     * @param limit
+     *            The maximum number of entities that may be assigned to the
+     *            parent entity.
+     * @param type
+     *            The type of the entities that shall be assigned.
+     * @param parentType
+     *            The type of the parent entity.
+     * @param countFct
+     *            Function to count the entities that are currently assigned to
+     *            the parent entity.
+     * 
+     * @throws QuotaExceededException
+     *             if the assignment operation would cause the quota to be
+     *             exceeded
+     */
+    public static void assertAssignmentQuota(final Long parentId, final long requested, final long limit,
+            @NotNull final String type, @NotNull final String parentType, final Function<Long, Long> countFct) {
+
+        // check if the quota is unlimited
+        if (limit <= 0) {
```

Comment:
check if it makes sense to have private helper methods isUnlimited, isExceeded...